src/libre/print/tree.c

```diff
@@ -57,6 +57,11 @@ fprintf_flags(FILE *f, enum ast_flags flags)
 	PR_FLAG(LAST,          "L");
 	PR_FLAG(UNSATISFIABLE, "X");
 	PR_FLAG(NULLABLE,      "0");
+	PR_FLAG(ANCHORED_START,"^");
+	PR_FLAG(ANCHORED_END,  "$");
+	PR_FLAG(END_NL,        "N");
+	PR_FLAG(CAN_CONSUME,   "c");
+	PR_FLAG(ALWAYS_CONSUMES, "C");
```

Comment:
Noting that we don't currently print `(struct ast_expr)->flags` from the corresponding AST rendering in src/libre/print/dot.c
(but we should!)

Reply:
We probably should, but I don't tend to use the dot view much -- since it doesn't always collapse down multiple edges between nodes, it's easy to produce dot views where nodes with 254 edges between them make the graph very large and difficult to read. If we rework dot output to handle that sort of thing better, we should include the flags at the same time.